modulus/datapipes/climate/climate_hdf5.py

```diff
@@ -0,0 +1,590 @@
+            self.logger.info(f"Climate file found: {data_path}")
+        self.n_years = len(self.data_paths)
+        self.logger.info(f"Number of years: {self.n_years}")
+
+        # get total number of examples and image shape from the first file,
+        # assuming other files have exactly the same format.
+        self.logger.info(f"Getting file stats from {self.data_paths[0]}")
+        with h5py.File(self.data_paths[0], "r") as f:
+            # truncate the dataset to avoid out-of-range sampling
+            data_samples_per_year = f["fields"].shape[0] - self.num_steps * self.stride
+            self.data_shape = f["fields"].shape[2:]
+
+            # If channels not provided, use all of them
+            if self.channels is None:
+                self.channels = [i for i in range(f["fields"].shape[1])]
+
+            # If num_samples_per_year use all
+            if self.num_samples_per_year is None:
+                self.num_samples_per_year = data_samples_per_year
+
+            # Adjust image shape if patch_size defined
+            if self.patch_size is not None:
+                self.cropped_data_shape = [
+                    s - s % self.patch_size[i] for i, s in enumerate(self.data_shape)
+                ]
+            else:
+                self.cropped_data_shape = self.data_shape
+            self.logger.info(f"Input data shape: {self.cropped_data_shape}")
+
+            # Get total length
+            self.total_length = self.n_years * self.num_samples_per_year
+
+            # Sanity checks
+            if max(self.channels) >= f["fields"].shape[1]:
+                raise ValueError(
+                    f"Provided channel has indexes greater than the number \
+                of fields {f['fields'].shape[1]}"
+                )
+
+            if self.num_samples_per_year > data_samples_per_year:
+                raise ValueError(
+                    f"num_samples_per_year ({self.num_samples_per_year}) > number of \
+                    samples available ({data_samples_per_year})!"
+                )
+
+            self.logger.info(f"Number of samples/year: {self.num_samples_per_year}")
+            self.logger.info(f"Number of channels available: {f['fields'].shape[1]}")
+
+    def _load_statistics(self) -> None:
+        """Loads climate statistics from pre-computed numpy files
+
+        The statistic files should be of name global_means.npy and global_std.npy with
+        a shape of [1, C, 1, 1] located in the stat_dir.
+
+        Raises
+        ------
+        IOError
+            If mean or std numpy files are not found
+        AssertionError
+            If loaded numpy arrays are not of correct size
+        """
+        # If no stats dir we just skip loading the stats
+        if self.stats_dir is None:
+            self.mu = None
+            self.std = None
+            return
+        # load normalisation values
+        mean_stat_file = self.stats_dir / Path("global_means.npy")
```

Comment:
Ideally, we should not hardcode the file names.

Reply:
Yeah, don't really like this either. I think it is best to leave this in for now though. The way we are dealing with normalization if very prone to error and I think explicitly making the file names like this makes it a little less likely someone will swap files or something...